patroni/postgresql/slots.py

```diff
@@ -332,6 +332,26 @@ def drop_replication_slot(self, name: str) -> Tuple[bool, bool]:
                             ' FULL OUTER JOIN dropped ON true'), name)
         return (rows[0][0], rows[0][1]) if rows else (False, False)
 
+    def _drop_physical_slot(self, name: str) -> bool:
+        """Drop a physical replication slot by name.
+
+        .. note::
+            If not able to drop the slot, it will log a message and set the flag to reload slots.
+
+        :param name: name of the slot to be dropped.
+        :returns: ``True`` if the slot was successfully dropped, ``False`` otherwise.
+        """
+        active, dropped = self.drop_replication_slot(name)
+        if dropped:
+            logger.info("Dropped physical replication slot '%s'", name)
```

Comment:
```suggestion
            logger.info("Dropped physical replication slot '%s'", name)
            if name in self._replication_slots:
                del self._replication_slots[name]
```

it will make everything a bit nicer.